<?xml version="1.0"?>
<!--
    This file is a wrapper around the original Tiago URDF.
    It includes the standard robot model and then adds a new
    fixed link for the fingertip frame.

    To use this, make sure to replace 'my_package' with the actual
    name of your ROS package.
-->
<robot name="tiago" xmlns:xacro="http://www.ros.org/wiki/xacro">
  <!-- Include the original TIAGo URDF. The path here is relative
       to the tiago_description package. We are specifically including
       the urdf.xacro file, which defines all the robot's components. -->
  <xacro:include filename="$(find tiago_description)/robots/tiago.urdf.xacro" />

  <!-- Define a new link and joint for the fingertip.
       This joint is fixed, so it doesn't move relative to the parent link.
       The parent link is the 'arm_tool_link', which is the end of the arm. -->
  <link name="fingertip_link" />

  <joint name="arm_tool_to_fingertip" type="fixed">
    <parent link="arm_tool_link" />
    <child link="fingertip_link" />
    <!-- The origin of the new fingertip_link is defined with an offset
         relative to the arm_tool_link.
         - 'xyz' defines the translation (in meters). You may need to
           adjust these values to match the precise location of your gripper's
           fingertip. A positive 'x' value moves it forward.
         - 'rpy' defines the rotation (in radians). -->
    <origin xyz="0.10 0 0" rpy="0 0 0" />
  </joint>

</robot>